cache/cache.go

```diff
@@ -38,21 +39,22 @@ type Func[T any] func() (T, error)
 // For performance, it's strongly recommended that you store pointers to objects
 // instead of actual objects.
 type Cache[T any] struct {
-	data        map[string]item[T]
+	// data is the actual internal cache storage.
+	data map[string]*cacheListItem[T]
+
+	// head points to the head of the linked list, tail points to the tail.
+	head, tail *cacheListItem[T]
```

Comment:
So, if we're trying to make a generalizable cache that is time-based, i question using a linked list. That is the typical implementation for LRUs, so i understand why it was added. But i would expect that in a truly generalizable time-based cache implementation, i would be able to specify different expiration times for different pieces of data. We don't necessarily have to implement that now, but if we wanted to support that, it wouldn't fit will with the linked-list based implementation, which would assume that items are in order of expiration.

Reply:
Yes, if we went with a generic TTL-based cache where individual items could have their own TTLs, we'd probably switch to a heap to make walking more efficient. I don't think we need that optimization here yet.